src/scripting_engine.c

```diff
@@ -0,0 +1,282 @@
+#include "scripting_engine.h"
+#include "dict.h"
+#include "functions.h"
+#include "module.h"
+
+typedef struct scriptingEngineImpl {
+    /* Engine specific context */
+    engineCtx *ctx;
+
+    /* Callback functions implemented by the scripting engine module */
+    engineMethods methods;
+} scriptingEngineImpl;
+
+typedef struct scriptingEngine {
+    sds name;                    /* Name of the engine */
+    ValkeyModule *module;        /* the module that implements the scripting engine */
+    scriptingEngineImpl *impl;   /* engine callbacks that allows to interact with the engine */
```

Comment:
Why is `impl` a separate allocation? It seems to be just a grouping of engineCtx and engineMethods. Why not do it without a separate allocation?

```suggestion
    scriptingEngineImpl impl;    /* engine context and callbacks to interact with the engine */
```

Reply:
Yeah, you're right.